src/main/java/io/ebeaninternal/server/type/DataBind.java

```diff
@@ -25,11 +29,30 @@
 
   private final StringBuilder bindLog = new StringBuilder();
 
+  private List<InputStream> streams;
+
+  private final BatchedPstmt batchedPstmt;
+
   private int pos;
 
+  /**
+   * Wraps a prepared statement. The statement will be closed on {@link DataBind#close()}.
+   */
   public DataBind(DataTimeZone dataTimeZone, PreparedStatement pstmt, Connection connection) {
     this.dataTimeZone = dataTimeZone;
     this.pstmt = pstmt;
+    this.streams = null;
+    this.batchedPstmt = null;
+    this.connection = connection;
+  }
+
+  /**
+   * Wraps a {@link BatchedPstmt} and takes accout of batch requirements.
+   */
+  public DataBind(DataTimeZone dataTimeZone, BatchedPstmt batchedPstmt, Connection connection) {
+    this.dataTimeZone = dataTimeZone;
+    this.pstmt = batchedPstmt.getStatement();
+    this.batchedPstmt = batchedPstmt;
```

Comment:
So, here comes the complex part - closing File-Streams - see the test case: https://github.com/ebean-orm/ebean/pull/1411/files#diff-b9f83866e73ea4adc1eceeaf35449eb6R112

So, what is done here:
1. DataBind will no longer close the prepared statement
2. DataBind will track a list of all InputStreams that are bound to the preparedStatement in non batch mode.
3. When the DataBind is closed, it will close all streams from the list (in non batch mode)
4. In batch mode, it delegates that list to the BatchedPstmt which handles the resource cleanup